rustaudit.go

```diff
@@ -193,3 +201,98 @@ func (x *machoExe) ReadRustDepSection() ([]byte, error) {
 
 	return depInfo.Data()
 }
+
+type wasmReader struct {
+	r io.ReaderAt
+}
+
+func (x *wasmReader) ReadRustDepSection() ([]byte, error) {
+	r := x.r
+	var offset int64 = 0
+
+	// Check the preamble (magic number and version)
+	buf := make([]byte, 8)
+	_, err := r.ReadAt(buf, offset)
+	offset += 8
+	if err != nil || !bytes.Equal(buf, wasmHeader) {
+		return nil, ErrUnknownFileFormat
+	}
+
+	// https://www.w3.org/TR/2019/REC-wasm-core-1-20191205/#custom-section%E2%91%A0
+	// Look through the sections until we find a custom .dep-v0 section or EOF
+	for {
+		// Read single byte section ID
+		sectionId := make([]byte, 1)
+		_, err = r.ReadAt(sectionId, offset)
+		offset += 1
+		if err == io.EOF {
+			return nil, ErrNoRustDepInfo
+		} else if err != nil {
+			return nil, ErrUnknownFileFormat
+		}
+
+		// Read section size
+		buf = make([]byte, 4)
+		_, err = r.ReadAt(buf, offset)
+		if err != nil {
+			return nil, ErrUnknownFileFormat
```

Comment:
nit: Is this the right error type to return here? Same for the reuse of this error type below.

Reply:
Leaving as is for client back-compatibility.